lib/Auth/Process/COmanageDbClient.php

```diff
@@ -324,62 +337,64 @@ public function process(&$state)
         try {
             assert('is_array($state)');
             if (isset($state['SPMetadata']['entityid']) && in_array($state['SPMetadata']['entityid'], $this->blacklist, true)) {
-                SimpleSAML_Logger::debug("[attrauthcomanage] process: Skipping blacklisted SP ". var_export($state['SPMetadata']['entityid'], true));
+                Logger::debug("[attrauthcomanage] process: Skipping blacklisted SP ". var_export($state['SPMetadata']['entityid'], true));
                 return;
             }
             if (empty($state['Attributes'][$this->userIdAttribute])) {
                 //echo '<pre>' . var_export($state, true) . '</pre>';
-                SimpleSAML_Logger::error("[attrauthcomanage] Configuration error: 'userIdAttribute' not available");
-                throw new SimpleSAML_Error_Exception(
+                Logger::error("[attrauthcomanage] Configuration error: 'userIdAttribute' not available");
+                throw new Error\Exception(
                     "attrauthcomanage configuration error: 'userIdAttribute' not available");
             }
             unset($state['Attributes']['uid']);
             $orgId = $state['Attributes'][$this->userIdAttribute][0];
-            SimpleSAML_Logger::debug("[attrauthcomanage] process: orgId=". var_export($orgId, true));
+            Logger::debug("[attrauthcomanage] process: orgId=". var_export($orgId, true));
             $basicInfo = $this->_getBasicInfo($orgId);
-            SimpleSAML_Logger::debug("[attrauthcomanage] process: basicInfo=". var_export($basicInfo, true));
-            if (empty($basicInfo['id']) || empty($basicInfo['status']) 
+            Logger::debug("[attrauthcomanage] process: basicInfo=". var_export($basicInfo, true));
+            if (empty($basicInfo['id']) || empty($basicInfo['status'])
                 || ($basicInfo['status'] !== 'A' && $basicInfo['status'] !== 'GP')) {
                   $state['UserID'] = $orgId;
-                  $params = array();
-                  $id = SimpleSAML_Auth_State::saveState($state, 'attrauthcomanage:register');
-                  $callback = SimpleSAML_Module::getModuleURL('attrauthcomanage/idp_callback.php', array('stateId' => $id));
-                  SimpleSAML_Logger::debug("[attrauthcomanage] process: callback url => " . $callback);
-                  $params = array("targetnew" => $callback);
+                  $params = [];
+                  $id = State::saveState($state, 'attrauthcomanage:register');
+                  $callback = Module::getModuleURL('attrauthcomanage/idp_callback.php', ['stateId' => $id)];
```

Comment:
Change `['stateId' => $id)];` to `['stateId' => $id]);`

Reply:
Done